numpy/ma/tests/test_core.py

```diff
@@ -4849,6 +4849,23 @@ def test_mask_or(self):
         cntrl = np.array([(1, (1, 1)), (0, (1, 0))], dtype=dtype)
         assert_equal(mask_or(amask, bmask), cntrl)
 
+    def test_allequal(self):
+        x = array([1, 2, 3], mask=[0, 0, 0])
+        y = array([1, 2, 3], mask=[1, 0, 0])
+        z = array([[1, 2, 3], [4, 5, 6]], mask=[[0, 0, 0], [1, 1, 1]])
+
+        assert allequal(x, y)
+        assert not allequal(x, y, fill_value=False)
+        assert allequal(x, z)
+
+        # test allequal for the same input, with mask=nomask, this test is for
+        # the scenario raised in https://github.com/numpy/numpy/issues/27201
+        assert allequal(x, x)
+        assert allequal(x, x, fill_value=False)
+
+        assert allequal(y, y)
+        assert not allequal(y, y, fill_value=False)
+
```

Comment:
@ianharris  As a test for `mask_or` I was thinking about something like this
```suggestion
    def test_mask_or(self):
        a = array([False, False])
        b = array([False, True])
        assert np.ma.mask_or(a, a) is numpy.ma.nomask # gh-27360
        np.testing.assert_array_equal(mask_or(a, b), np.array([False,  True]))    
```
There are no explicit tests for `mask_or` (that I can find), but it is a public method so a test would be good

Reply:
Cool, that makes sense. I found a set of existing `mask_or` tests, so I added your `mask_or(a, a) is nomask` case there.